### ISSUE TITLE
Migrate docs site to material for mkdocs

### ISSUE THREAD

**Issue by thobson88** [Living-with-machines/T-Res#284]
Partially done on branch `mkdocs`, and merged into `276-refactor` to capture changes outside the `docs/` directory (mostly updated docstrings). To be continued on another feature branch after #286.

TODO:
 - [x] Set up MkDocs site and translate from Sphinx version
 - [x] Fix CI
 - [x] Docs homepage
 - [ ] Getting Started section
 - [x] Reference section
 - [ ] API section
 - [ ] Experiments section
 - [ ] Class diagrams & explanation for geoparser modules
 - [ ] Instructions for running granular tests
